### PR TITLE
Suppress pointless warnings coming from SBT

### PR DESCRIPTION
**Pull request by hubertp**
[ci no changelog needed]

### Pull Request Description

This change introduces a custom LogManager for console that allows for
excluding certain log messages. The primarily reason for introducing
such LogManager/Appender is to stop issuing hundreds of pointless
warnings coming from the analyzing compiler (wrapper around javac) for
classes that are being generated by annotation processors.

The output looks like this:
```
[info] Cannot install GraalVM MBean due to Failed to load org.graalvm.nativebridge.jni.JNIExceptionWrapperEntryPoints
[info] compiling 129 Scala sources and 395 Java sources to /home/hubert/work/repos/enso/enso/engine/runtime/target/scala-2.13/classes ...
[warn] Unexpected javac output: warning: File for type 'org.enso.interpreter.runtime.type.ConstantsGen' created in the last round will not be subject to annotation processing.
[warn] 1 warning.
[info] [Use -Dgraal.LogFile=<path> to redirect Graal log output to a file.]
[info] Cannot install GraalVM MBean due to Failed to load org.graalvm.nativebridge.jni.JNIExceptionWrapperEntryPoints
[info] foojavac Filer
[warn] Could not determine source for class org.enso.interpreter.node.expression.builtin.number.decimal.CeilMethodGen
[warn] Could not determine source for class org.enso.interpreter.node.expression.builtin.resource.TakeNodeGen
[warn] Could not determine source for class org.enso.interpreter.node.expression.builtin.error.ThrowErrorMethodGen
[warn] Could not determine source for class org.enso.interpreter.node.expression.builtin.number.smallInteger.MultiplyMethodGen
[warn] Could not determine source for class org.enso.interpreter.node.expression.builtin.warning.GetWarningsNodeGen
[warn] Could not determine source for class org.enso.interpreter.node.expression.builtin.number.smallInteger.BitAndMethodGen
[warn] Could not determine source for class org.enso.interpreter.node.expression.builtin.error.ErrorToTextNodeGen
[warn] Could not determine source for class org.enso.interpreter.node.expression.builtin.warning.GetValueMethodGen
[warn] Could not determine source for class org.enso.interpreter.runtime.callable.atom.AtomGen$MethodDispatchLibraryExports$Cached
....
```

The output now has over 500 of those and there will be more. Much more
(generated by our and Truffle processors).
There is no way to tell SBT that those are OK. One could potentially
think of splitting compilation into 3 stages (Java processors, Java and
Scala) but that will already complicate the non-trivial build definition
and we may still end up with the initial problem.
This is a fix to make it possible to get reasonable feedback from
compilation without scrolling mutliple screens *every single time*.

Also fixed a spurious warning in javac processor complaining about
creating files in the last round.

Related to https://www.pivotaltracker.com/story/show/182138198

### Checklist

Please include the following checklist in your PR:

- [ ] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run.sh ide dist` and `./run.sh ide watch`.

### CODE REVIEW

**Review by mwu-tow — APPROVED** [review]
YES

File: lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/TypeProcessor.java [hunk 1]
```diff
@@ -53,6 +55,17 @@ protected boolean handleProcess(
     for (TypeElement annotation : annotations) {
       Set<? extends Element> annotatedElements = roundEnv.getElementsAnnotatedWith(annotation);
       for (Element elt : annotatedElements) {
+        if (jfo == null) {
+          // Create generator for Java source file once(!) so that it can be used at the last
+          // round of processing. Otherwise, javac complains that the generated file won't be
+          // used in further processing. That's fine, we know it won't.
+          try {
+            jfo = processingEnv.getFiler().createSourceFile(ConstantsGenFullClassname);
```
Comment (4e6) [inline lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/TypeProcessor.java]: Why are we setting `jfo` in a loop, and not in the class constructor for example?

Comment (hubertp) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/TypeProcessor.java]: It's not particularly happy about it
```
[error] error: Bad service configuration file, or exception thrown while constructing Processor object: javax.annotation.processing.Processor: Provider org.enso.interpreter.dsl.TypeProcessor could not be instantiated
```

Comment (hubertp) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/TypeProcessor.java]: Just to maybe expand on it, it's particularly happy about using the processing env in the constructor, it seems

Comment (4e6) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/TypeProcessor.java]: I see, the usual initialization issues. But can we maybe set it once in this function, just outside of loop?

**Review by 4e6 — APPROVED** [review]
All good! Just a comment about the initialization.

File: lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/TypeProcessor.java (hunk 1 above)
Comment (hubertp) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/TypeProcessor.java]: :+1:

**Pull request merged**